sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala

```diff
@@ -412,8 +417,13 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
    */
   override def dropGlobalTempView(viewName: String): Boolean = {
     sparkSession.sessionState.catalog.getGlobalTempView(viewName).exists { viewDef =>
-      sparkSession.sharedState.cacheManager.uncacheQuery(
-        sparkSession, viewDef, cascade = false)
+      try {
+        val plan = sparkSession.sessionState.executePlan(viewDef)
+        sparkSession.sharedState.cacheManager.uncacheQuery(
+          sparkSession, plan.analyzed, cascade = false)
+      } catch {
+        case NonFatal(_) => // ignore
```

Comment:
is it safer to not add this try-catch? if there are failures, then the uncache fails and we should not swallow the exception?

Reply:
This is also discussed above. Suppose we have:
```sql
CREATE TEMPORARY VIEW v1 AS SELECT * FROM v2
```
a Spark user can drop v2 first followed by v1. In this case the `uncacheQuery` will fail because `v2` is already gone. Consequently, the view will not be dropped. This seems to be a quite common scenario. 

Also this is [following the `DropTableCommand](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala#L239)`, which swallows any non-fatal exception and proceed to drop the view.

Reply:
thanks for the explanation!